### PR TITLE
Refactor notifyMentionedUsers method

### PR DESCRIPTION
**Pull request by vitormattos**

### CODE REVIEW

**Comment by vitormattos** [comment]
I identified that after the refactor reduced a bit the quantity of queries on this flow.

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: lib/Chat/Notifier.php [hunk 1]
```diff
@@ -95,38 +95,69 @@ public function __construct(INotificationManager $notificationManager,
...
+			return !in_array($user['id'], $alreadyNotifiedUsers);
+		});
+	}
+
+	private function addMentionAllToList(Room $chat, array $list): array {
+		$usersToNotify = array_filter($list, function (array $user): bool {
+			return $user['id'] !== 'all';
+		});
+
+		if (count($list) === count($usersToNotify)) {
```
[the remark this answers is not in the document]
Comment (vitormattos) [reply lib/Chat/Notifier.php:138]: If I do this change the all will be stay on array and the method getActorsByType will do a query using all as a username and all will be returned by the method. The All isn't a user to notify.

File: lib/Chat/Notifier.php [hunk 2]
```diff
@@ -257,20 +294,31 @@ public function markMentionNotificationsRead(Room $chat, ?string $userId): void
...
-				$userIds[] = $mention['id'];
+				$mentionedUsers[] = [
+					'id' => $mention['id'],
+					'type' => 'users'
+				];
 			}
 		}
-
-		return $userIds;
+		return $mentionedUsers;
```
[the remark this answers is not in the document]
Comment (vitormattos) [reply lib/Chat/Notifier.php:321]: Not, because I included the type on array.

[the remark this answers is not in the document]
Comment (vitormattos) [reply lib/Chat/Notifier.php:320]: Will be necessary array_filter and array_map using two interactions on the array. Using foreach I only used one interaction.

[the remark this answers is not in the document]
Comment (vitormattos) [reply lib/Chat/Notifier.php:320]: The type returned on mention array have the type on singular, I used this step to use the right type used on Talk.

File: lib/Chat/Notifier.php [hunk 3]
```diff
@@ -95,38 +95,69 @@ public function __construct(INotificationManager $notificationManager,
...
+
+	private function removeAlreadyNotifiedUsers(array $usersToNotify, array $alreadyNotifiedUsers): array {
+		return array_filter($usersToNotify, function (array $user) use ($alreadyNotifiedUsers): bool {
+			return !in_array($user['id'], $alreadyNotifiedUsers);
+		});
+	}
+
+	private function addMentionAllToList(Room $chat, array $list): array {
+		$usersToNotify = array_filter($list, function (array $user): bool {
+			return $user['id'] !== 'all';
```
[the remark this answers is not in the document]
Comment (vitormattos) [reply lib/Chat/Notifier.php:136]: The objective isn't check if `@all` exists, the objective is remove `@all`

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**